Core/clim-basic/input.lisp

```diff
@@ -171,22 +172,17 @@ use condition-variables nor locks."))
 
 (defmethod event-queue-read-with-timeout ((queue simple-event-queue) timeout wait-function)
   (do-port-force-output queue)
-  (when wait-function
-    (warn "EVENT-QUEUE-READ-WITH-TIMEOUT: ignoring WAIT-FUNCTION."))
   (let ((port (event-queue-port queue))
         (timeout-time (and timeout (+ timeout (now)))))
     ;; timeout-time may be past-due already but we may have an event waiting to be
     ;; processed for some time. Do preemptive check. -- jd 2018-12-30
     (process-next-event port :timeout 0)
-    (loop
-       as decay = (compute-decay timeout-time (event-schedule-time queue))
-       do
-         (when decay (maxf decay 0))
-         (if-let ((event (%event-queue-read queue)))
-           (return event)
-           (if (and timeout-time (> (now) timeout-time))
-               (return nil)
-               (process-next-event port :wait-function wait-function :timeout decay))))))
+    (let ((decay (compute-decay timeout-time (event-schedule-time queue))))
```

Comment:
it might have been a leftover of looping in concurrent-event-queue which is looping to catch on spurious CV wakeups which I have corrected. I'm not sure at this point of time.